libqtile/layout/floating.py

```diff
@@ -259,10 +259,12 @@ def configure(self, client, screen_rect):
             bw,
             bc,
             above,
+            z=None
         )
         client.unhide()
 
     def add(self, client):
+        client.z = 0 if len(self.clients) == 0 else max(self.clients, key=lambda c: c.z).z + 1
```

Comment:
I wasn't sure what was fastest so I tested this keyed max() call against list comprehension:
```
import random
import timeit

class A:
    def __init__(self, z):
        self.z = z

clients = [A(i) for i in range(10)]
random.shuffle(clients)

print(timeit.timeit(
    'max(clients, key=lambda c: c.z).z', globals={'clients': clients}, number=1000000
))
print(timeit.timeit(
    'max([c.z for c in clients])', globals={'clients': clients}, number=1000000
))
```
Which gave (in seconds):
```
1.3131903419998707
0.8532288009992044
```
So it's close, but looks like list comprehension here wins the speed test.

Reply:
I'll change it, thanks for the test. Although we are talking about 0.5 microseconds whenever a window is added.^^